documentation/concepts/check.md

```diff
@@ -4,25 +4,88 @@ menu: check
+    <td><strong>Command &amp; Flag</strong></td>
+    <td><strong>Output</strong></td>
+</tr>
+<tr>
+    <td><em>In beta</em><br>Will these changes have the effect I am expecting?</td>
+    <td>check <b>-changes</b></td>
+    <td>ChangeReport.html,<br>ChangeReport.json</td>
+</tr>
+
+<tr>
+    <td><em>In beta</em><br>What database changes have been made recently?</td>
+    <td>check <b>-changes</b></td>
+    <td>ChangeReport.html,<br>ChangeReport.json</td>
+</tr>
+
+<tr>
+    <td><em>Coming soon</em><br>Is the production database in the state I am expecting it to be in?</td>
+    <td>check <b>-drift</b></td>
+    <td>&nbsp;</td>
+</tr>
+<tr>
+    <td><em>Coming soon</em><br>Are our changes following internal policies?</td>
+    <td>check <b>-drift</b></td>
+    <td>&nbsp;</td>
+</tr>
+</table>
+
+## `Check –changes` 
+### Overview 
+the `–changes` flag produces a report indicating differences between your target database (ie. the one you want to change) and what your migration scripts define (ie. the set of instructions you want to use to change your target database).  
+You can use this capability pre- and post-deployment: 
+- In pre-deployment scenarios to check the effect of your pending changes 
+- In post-deployment scenarios to capture a history of changes for retrospective auditing or reporting 
+
+In either scenario, using the `-changes` flag will help you infer which database objects will be/have been affected - and how – when you execute/have executed your migration script(s). 
+
+### Requirements and behaviour
+The `check –changes` command and flag works by building a temporary database. This temporary database is first made to reflect the state of your target schema, and then made to reflect your target schema with the pending changes applied.  
+
+The difference between the two states of this temporary database (target now, and target with changes applied) represents the effect your pending migrations will have (or have had) when the scripts are (or were) executed. This difference is captured as an artefact called a “Change Report”. The change report is available as both HTML (human readable) and JSON (machine readable) formats. 
+
+The process works like this: 
+
+1. Specify your target database location 
+    1. This is the database you want to apply your changes to, where Flyway is already being used to manage migrations (i.e. A Flyway migrations table exists)) 
+1. Specify a temporary database, either  
+    1. A location for the temporary database (Flyway will create the database for you), or 
+    1. An existing temporary database (note: Flyway will “clean” this database, so if you specify a full database, you must ensure it is ok to for Flyway to erase its schema) 
+1. Run `flyway check –changes` 
+
+Flyway’s `check –changes` will then: 
+1. Clean your temporary database 
+1. Check the latest version of your target database (let’s say it’s at V4) 
+1. Migrate the temporary database to the same version as your target database 
```

Comment:
Implementation-wise these two steps have been modified, since migrating to a target doesn't account for migrations undone out of order. For example, if the current version is V4 but V3 was undone, migrating just to V4 would also apply V3, which is incorrect

The current implementation queries the target DB for the list of applied migrations, and applies them with the use of `cherryPick` to ensure just the right ones are applied

Reply:
Changed the language so it doesn't imply a simple linear path